### ISSUE TITLE
About training set mentioned in paper

### ISSUE THREAD

**Issue by xuyifan-0731** [google-research/android_world#235]
Hi!  

In the *AndroidWorld* paper, you mentioned:  

> "Beyond testing agent robustness, the dynamic construction of tasks supports the use of online learning methodologies, particularly reinforcement learning [45, 29, 19, 14]. It also simplifies the generation of distinct train/test datasets, facilitating supervised learning experiments [19, 44, 12]."  

However, in the code repository, I couldn't find any methods on how to utilize the existing content to construct training data.  

Does this mean that simply modifying each task's `_TASK_RANDOM_SEED` is sufficient for constructing training prompts, along with the corresponding app state and success judgment?

**Comment by crawles** [comment]
Hi, thanks for reaching out. One path you could pursue to collect training
data is to collect trajectories for a given seed. Note this requires using
humans or agents. Then you could test a new agent on different seeds. We do
not provide any trajectories for this, so you would have to DIY.

Chris

**Comment by xuyifan-0731** [comment]
Thank you for your explanation! Using an agent to collect data for tasks replacing the seed is a feasible approach. However, I still have some concerns: are there certain tasks where replacing the seed does not change the task? 

For example, in some system-related tasks, I found that `generate_random_params` does not generate a random number but produces a fixed result. Would using such rollout data be considered test set leakage?

An example:

```python
class SystemBluetoothTurnOffVerify(_SystemBluetoothToggle):
  """Task for verifying that Bluetooth is already turned off.

  Precondition: Bluetooth is off.
  """

  def initialize_task(self, env: interface.AsyncEnv) -> None:
    super().initialize_task(env)
    adb_utils.toggle_bluetooth(env.controller, 'off')

  @classmethod
  def generate_random_params(cls) -> dict[str, str]:
    return {'on_or_off': 'off'}

**Comment by crawles** [comment]
Ah yes, that is true. However, the large majority contain tasks that have
random parameters. Unfortunately I don’t have a list but you could check
pretty easily by comparison different seeds for each task

Chris

**Comment by xuyifan-0731** [comment]
thanks a lot
